README.md

```diff
@@ -1 +1,90 @@
-# breakers
\ No newline at end of file
+# Breakers
+
+Breakers is a Ruby gem that implements the circuit breaker pattern for Ruby using a Faraday middleware. It is designed to handle the case
+where your app communicates with one or more backend services over HTTP and those services could possibly go down. Data about the success
+and failure of requests is recorded in Redis, and the gem uses this to determine when an outage occurs. While a service is marked as down,
+requests will continue to flow through every minute to check if it has returned to being alive.
+
+## Installation
+
+Add this line to your application's Gemfile:
+
+```ruby
+gem 'breakers'
+```
+
+And then execute:
+
+    $ bundle
+
+Or install it yourself as:
+
+    $ gem install breakers
+
+## Usage
+
+### Getting Connected
+
+The gem allows you to define your services like this:
+
+```ruby
+service = Breakers::Service.new(name: 'messaging') do |request_env|
+  request_env.uri.host =~ /.*messaging\.va\.gov/
+end
+```
+
+The name parameter is used for logging and reporting only. On each request, the block will be called with the request's environment, and
+the block should return true if the service applies to it.
+
+A Breakers::Client is the basic data structure for accessing the state and creating connections. It requires a redis connection and one or
```

Comment:
It uses some specific functionality of redis, like ZSETs for being able to pull thing in order. Perhaps I could engineer that away, but it seems like it would necessarily make it less efficient. The code doesn't actually create a Redis connection, so it works fine with not having it as a dependency. The client of the gem would create the connection, but as long as they pass me something that implements the ruby-redis gem's interface it doesn't matter to me.